### PR TITLE
Feat(Gmx): Gmx swap

### PR DESCRIPTION
**Pull request by Quazia**
This is blocked by https://github.com/rabbitholegg/questdk/pull/14

### CODE REVIEW

**Review by jimobrien — APPROVED** [review]
LGTM - will make another pass once the GMX v2 is worked in!

File: packages/gmx/src/GMX.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const response = await fetch('https://api.gmx.io/tokens')
+
+    // Check if the request was successful (status code 200)
+    if (response.ok) {
+      // Parse the JSON response into a JavaScript object
+      const data = (await response.json()) as Array<{ data: any; id: string }>
+      return data.map((token) => token.id)
+    } else {
+      console.error(`Request failed with status code: ${response.status}`)
+      return []
```
Comment (jimobrien) [inline packages/gmx/src/GMX.ts:44]: Should we fail over to a static list of the majors?

Comment (Quazia) [reply packages/gmx/src/GMX.ts:95]: Great idea - I've added sensible defaults to a static list equivalent to what the API currently returns.

File: packages/gmx/src/contract-addresses.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export const GMX_ROUTERV1_ADDRESS = '0xabbc5f99639c9b6bcb58544ddf04efa6802f4064'
+export const GMX_ROUTERV2_ADDRESS = '0x3B070aA6847bd0fB56eFAdB351f49BBb7619dbc2'
+export const DEFAULT_TOKEN_LIST_URL = [
```
Comment (jimobrien) [inline packages/gmx/src/contract-addresses.ts:3]: should be able to do `const DEFAULT_TOKEN_LIST_URL: Address[] = [ ... ]` and you shouldn't have to do the type casting other places 

`Address` comes from viem

File: packages/gmx/src/GMX.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+export const getSupportedTokenAddresses = async (_chainId: number) => {
+  try {
+    // Send a GET request to the specified URL using the fetch API
+    const response = await fetch('https://api.gmx.io/tokens')
+
+    // Check if the request was successful (status code 200)
+    if (response.ok) {
+      // Parse the JSON response into a JavaScript object
+      const data = (await response.json()) as Array<{ data: any; id: string }>
+      return data.map((token) => token.id) as `0x${string}`[]
```
Comment (jimobrien) [inline packages/gmx/src/GMX.ts:92]: ` import { Address } from 'viem'`

`as Address`

File: packages/gmx/src/GMX.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    } else {
+      console.error(`Request failed with status code: ${response.status}`)
+      return []
+    }
+  } catch (error) {
+    console.error(
+      `An error occurred: ${(error as { message: string }).message}`,
+    )
+    return DEFAULT_TOKEN_LIST_URL as `0x${string}`[]
```
Comment (jimobrien) [inline packages/gmx/src/GMX.ts:101]: See https://github.com/rabbitholegg/questdk-plugins/pull/28/files#r1333904567

File: packages/gmx/src/GMX.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    })
+  }
+  return compressJson({
+    to: contractAddress, // The contract address of the swap
+    chainId: chainId, // The chain id of the swap
+    input: {
+      $abi: GMX_SWAPV2_ABI,
+      data: {
+        $some: {
+          $abi: GMX_SWAPV2_ABI,
```
Comment (jimobrien) [inline packages/gmx/src/GMX.ts:66]: 👏 👏 👏

**Review by jimobrien — APPROVED** [review]
Some nits on typing the addresses differently but otherwise LGTM 

Nice work!

**Pull request merged**